kong/plugins/ai-proxy/schema.lua

```diff
@@ -0,0 +1,146 @@
+  }
+}
+
+local model_options_schema = {
+  description = "Key/value settings for the model",
+  type = "record",
+  required = false,
+  fields = {
+    { max_tokens = { type = "integer", description = "Defines the max_tokens, if using chat or completion models.", required = false }},
+    { temperature = { type = "number", description = "Defines the matching temperature, if using chat or completion models.", required = false, between = { 0.0, 5.0 } } },
+    { top_p = { type = "number", description = "Defines the top-p, if supported.", required = false, between = { 0 ,1 } } },
+    { top_k = { type = "integer", description = "Defines the top-k, if supported.", required = false }},
+    { anthropic_version = { type = "string", description = "Defines the schema/API version, if using Anthropic provider.", required = false } },
+    { azure_instance = { type = "string", description = "Instance name for Azure OpenAI hosted models.", required = false }},
+    { llama2_format = { type = "string", description = "If using llama2 provider, select the upstream message format.", required = false, one_of = { "raw", "openai"} } },
+    { upstream_url = typedefs.url { description = "Manually specify or override the full URL to the AI operation endpoints, when calling (self-)hosted models, or for running via a private endpoint.", required = false } },
+  }
+}
+
+local model_schema = {
+  type = "record",
+  required = true,
+  fields = {
+    { provider = { type = "string", description = "AI provider request format - Kong translates requests to and from the specified backend compatible formats.", required = true, one_of = { "openai", "azure", "anthropic", "cohere", "mistral", "llama2", "preserve" }, default = "openai" } },
+    { name = { type = "string", description = "Model name to execute.", required = true, } },
+    { options = model_options_schema },
+  }
+}
+
+local logging_schema = {
+  type = "record",
+  required = true,
+  fields = {
+    { log_statistics = { type = "boolean", description = "If enabled and supported by the driver, will add model usage and token metrics into the Kong log plugin(s) output.", required = true, default = true } },
+    { log_payloads = { type = "boolean", description = "If enabled, will log the request and response body into the Kong log plugin(s) output.", required = true, default = false } },
+  }
+}
+
+return {
+  name = "ai-proxy",
+  fields = {
+    { protocols = typedefs.protocols_http },
+    { consumer = typedefs.no_consumer },
+    { config = {
+      type = "record",
+      fields = {
+        { route_type = { type = "string", description = "The model's operation implementation, for this provider.", required = true, one_of = { "llm/v1/chat", "llm/v1/completions" } } },
+        { auth = auth_schema },
+        { model = model_schema },
+        { logging = logging_schema },
```

Comment:
question: why is this not part of a separate logging plugin?

Reply:
Because there is specific request and response information that I need from the API call in order to fulfil monitoring requirements. It feeds the data into the normal `kong.log` object, which is the serialized and written to whatever the user chooses to write to, e.g. file-log, http-log.

I am not sure this is very easy to do with a separate plugin, but we can discuss that!

Reply:
So there is no LLM-specific logging plugin and the existing access log plugins are re-used?

Reply:
Yes. To avoid the situation where sooner than later we are asked to basically re-create every logging plugin we already have, but specifically for AI Proxy. 

So we decided to augment the existing logs instead, and leverage the existing logging plugins for the actual log storage/sending.

Reply:
Good.
That may highlight other shortcomings of Kong. More on that some other time if they come up.